### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by PastushenkoVik**
https://pastushenkovik.github.io/react_films-list-use-raw-redux/

файлы в комите Solution

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,79 +1,100 @@
...
-      filmsList: [
-        ...prevState.filmsList,
-        {
-          id: prevState.filmsList[prevState.filmsList.length - 1].id + 1,
-          ...newFilm,
-        },
-      ],
+  componentWillUnmount() {
+    this.unSubscribe();
+  }
```
Comment (irenhh) [inline src/App.js]: you don't need these methods here

File: src/App.js [hunk 2]
```diff
@@ -1,79 +1,100 @@
...
+      imdbUrl,
     }));
   };
 
-  handleSearchChange = ({ target }) => {
-    this.setState({ searchWord: target.value });
-  };
+  handleSearchChange = ({ target }) => this.setState({
+    searchWord: target.value,
+    error: null,
```
Comment (irenhh) [inline src/App.js]: error has to be in the store

File: src/App.js [hunk 3]
```diff
@@ -1,79 +1,100 @@
...
+
+      store.dispatch(addNewFilm({
+        id: imdbID,
+        title: Title,
+        description: Plot,
+        imgUrl: Poster,
+        imdbUrl: Website,
+      }));
+    } catch (error) {
+      this.setState({ error: error.message });
```
Comment (irenhh) [inline src/App.js]: if you use it for `throw new error`, then remove it

File: src/App.js [hunk 4]
```diff
@@ -93,21 +114,16 @@ export class App extends Component {
...
                 path="/film/:id"
-                render={({ match }) => {
-                  const film = filmsList
-                    .find(f => String(f.id) === match.params.id);
-
-                  return (
-                    <FilmDetails {...film} />
-                  );
-                }}
+                component={({ match }) => (
```
Comment (irenhh) [inline src/App.js]: just use link to component

**Review by irenhh — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 4 above)
Comment (irenhh) [reply src/App.js]: you didn't make change according to this comment

**Review by irenhh — CHANGES_REQUESTED** [review]

**Review by danheim — APPROVED** [review]

**Pull request closed without merging**